Remove global data-dir flag

trawler opened the pull request:
Signed-off-by: Karen Almog <kalmog@mirantis.com>

## Description

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Fixes #942

## Type of change

`data-dir` is already used as a local flag under controller & worker options.
This PR removed `data-dir` and `config` as a global flag under the root command (and therefore removing it from other non-related subcommands.

<!-- check the related options -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Manual test
- [ ] Auto test added

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

trawler commented:
seems that tests are failing due to the broken `--debug` flag, that is expected to be fixed in #1659

jnummelin commented on cmd/root.go:
```diff
@@ -107,10 +107,6 @@ func NewRootCmd() *cobra.Command {
 	cmd.Long = longDesc
-
-	// workaround for the data-dir location input for the kubectl command
-	cmd.PersistentFlags().AddFlagSet(config.GetKubeCtlFlagSet())
```
Don't quite get how this solves the original issue, care to explain a bit?

trawler replied:
The name is a bit misleading (not sure why it was named like that originally) but `GetKubeCtlFlagSet` contains the global flags data-dir and debug (https://github.com/k0sproject/k0s/blob/f639051d29e31b7d7f6b31dcd30171fb882a6051/pkg/config/cli.go#L121-L126).
By removing it from from root.go, we basically ensure that only commands that need these flags include them. When this was in the root.go, these flags were inherited by all subcommands, which we don't want them to.

twz123 reviewed (approved):
This is somehow a breaking change. Are we okay with this? (Maybe mention this on the GitHub release page)

trawler commented:
> This is somehow a breaking change. Are we okay with this? (Maybe mention this on the GitHub release page)

It's actually not so much of a breaking change, more of a bug fix. The flags that are now removed were not actually used at all by the commands that didn't need them, so they were just unnecessary flags that are now removed.
